packages/langium/src/workspace/document-builder.ts

```diff
@@ -79,6 +80,25 @@ export interface DocumentBuilder {
      * Notify the given callback when a set of documents has been built reaching a desired target state.
      */
     onBuildPhase(targetState: DocumentState, callback: DocumentBuildListener): Disposable;
+
+    /**
+     * Wait until the workspace has reached the specified state for all documents.
+     *
+     * @param state The desired state. The promise won't resolve until all documents have reached this state
+     * @param cancelToken Optionally allows to cancel the wait operation, disposing any listeners in the process
+     * @throws `OperationCancelled` if cancellation has been requested before the state has been reached
+     */
+    waitUntil(state: DocumentState, cancelToken?: CancellationToken): Promise<void>;
+
+    /**
+     * Wait until the document specified by the {@link uri} has reached the specified state.
+     *
+     * @param state The desired state. The promise won't resolve until the document has reached this state
+     * @param uri The specified URI that points to the document. If the URI does not exist, the promise will resolve once the workspace has reached the specified state.
```

Comment:
I just stumbled over
> If the URI does not exist, the promise will resolve once the workspace has reached the specified state.

Would it be possible to reflect that in the result type, like `Promise<URI | undefined>` returning the URI if a corresponding document exists?